agents/builtin/mongodb/internal/profiler/profiler_test.go

```diff
@@ -15,3 +15,81 @@
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
 package profiler
+
+import (
+	"testing"
+	"time"
+
+	"github.com/percona/pmgo"
+	"github.com/percona/pmm/api/qanpb"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/mgo.v2/bson"
+
+	"github.com/percona/pmm-agent/agents/builtin/mongodb/internal/profiler/aggregator"
+	"github.com/percona/pmm-agent/agents/builtin/mongodb/internal/report"
+	"github.com/percona/pmm-agent/agents/builtin/mongodb/internal/test/profiling"
+)
+
+func TestProfiler(t *testing.T) {
+	aggregator.DefaultInterval = time.Duration(5 * time.Second)
+	url := "mongodb://pmm-agent:root-password@127.0.0.1:27017"
+	p := profiling.New(url)
+
+	logrus.SetLevel(logrus.DebugLevel)
+	dialInfo, err := pmgo.ParseURL(url)
+	fatalOnErr(t, err)
+
+	dialer := pmgo.NewDialer()
+
+	sess, err := createSession(dialInfo, dialer)
+	fatalOnErr(t, err)
+
+	err = sess.DB("test").DropDatabase()
+	fatalOnErr(t, err)
+
+	err = p.Enable("test")
+	fatalOnErr(t, err)
+
+	ms := &testWriter{t: t}
+	prof := New(dialInfo, dialer, logrus.WithField("component", "profiler-test"), ms, "test-id")
+	err = prof.Start()
+	fatalOnErr(t, err)
+
+	err = sess.DB("test").C("peoples").Insert(bson.M{"name": "Anton"}, bson.M{"name": "Alexey"})
+	fatalOnErr(t, err)
+
+	<-time.After(aggregator.DefaultInterval)
+
+	err = prof.Stop()
+	fatalOnErr(t, err)
+
+	err = p.Disable("test")
+	fatalOnErr(t, err)
+}
+
+type testWriter struct {
+	t *testing.T
+}
+
+func (tw *testWriter) Write(actual *report.Report) error {
+	assert.NotNil(tw.t, actual)
```

Comment:
That should be `require.NotNil`, otherwise `actual.Buckets` below will panic.
There are other similar places where `require` should used.